### PR TITLE
ECOM-7410 Update course seat validation check to prevent product title from having HTML

### PR DESCRIPTION
**Pull request by Ayub-Khan**
**ECOM-7410**

**Description :**
Students were unable to checkout using cybersource when product title had HTML in it. Updated the course seat creation validation check for product title to check for HTML.

**Testing :**
Locally tested changes Working fine. 
All tests are passing locally.

### CODE REVIEW

**Review by clintonb — CHANGES_REQUESTED** [review]
We should fix this at the source—CAT. Don't allow courses to be created with invalid text.

Also, please include the ticket number on the last line of your commit message.

**Review by attiyaIshaque — CHANGES_REQUESTED** [review]
Code changes look good. Could you add a regression test for this?

File: ecommerce/static/js/utils/validation_patterns.js [hunk 1]
```diff
@@ -14,5 +14,13 @@ define([
         _.extend(Backbone.Validation.messages, {
             courseId: gettext('The course ID is invalid.')
         });
+
+        _.extend(Backbone.Validation.patterns, {
+            productName: /^((?!&\w+;|<\/*\w+>).)*$/
+        });
+
+        _.extend(Backbone.Validation.messages, {
+            productName: gettext('The product name is invalid.')
```
Comment (clintonb) [inline ecommerce/static/js/utils/validation_patterns.js]: Provide helpful text that explains what is considered valid. Also, run `make fake_translations` to update the translations.

**Review by clintonb — CHANGES_REQUESTED** [review]
This needs tests.

**Comment by Ayub-Khan** [comment]
@clintonb @attiyaIshaque   updated

File: ecommerce/static/js/utils/validation_patterns.js [hunk 2]
```diff
@@ -14,5 +14,13 @@ define([
         _.extend(Backbone.Validation.messages, {
             courseId: gettext('The course ID is invalid.')
         });
+
+        _.extend(Backbone.Validation.patterns, {
+            productName: /^((?!&\w+;|<\/*\w+>).)*$/
+        });
+
+        _.extend(Backbone.Validation.messages, {
+            productName: gettext('The product name contains HTML in it.')
```
Comment (clintonb) [inline ecommerce/static/js/utils/validation_patterns.js]: The product name cannot contain HTML.

**Review by clintonb — APPROVED** [review]
Remember to squash your commits, and move the ticket number to the last line of your commit message.

**Review by attiyaIshaque — APPROVED** [review]
Once commits squashed 👍

**Pull request merged**